cloud/src/meta-service/meta_service_partition.cpp

```diff
@@ -276,20 +319,27 @@ void MetaServiceImpl::drop_index(::google::protobuf::RpcController* controller,
     std::string to_save_val;
     {
         RecycleIndexPB pb;
+        pb.set_db_id(request->db_id());
         pb.set_table_id(request->table_id());
         pb.set_creation_time(::time(nullptr));
         pb.set_expiration(request->expiration());
         pb.set_state(RecycleIndexPB::DROPPED);
         pb.SerializeToString(&to_save_val);
     }
     bool need_commit = false;
+    DropIndexLogPB drop_index_log;
+    drop_index_log.set_db_id(request->db_id());
+    drop_index_log.set_table_id(request->table_id());
+    drop_index_log.set_expiration(request->expiration());
+
     for (auto index_id : request->index_ids()) {
         auto key = recycle_index_key({instance_id, index_id});
         std::string val;
         err = txn->get(key, &val);
         if (err == TxnErrorCode::TXN_KEY_NOT_FOUND) { // UNKNOWN
             LOG_INFO("put recycle index").tag("key", hex(key));
             txn->put(key, to_save_val);
+            drop_index_log.add_index_ids(index_id);
```

Comment:
It appears that the recycle index key should not be written if versioned writes are enabled.

The drop_partition implementation is wrong.